odl/solvers/functional/functional.py

```diff
@@ -500,7 +501,7 @@ def __init__(self, func, scalar):
             raise TypeError('`func` {!r} is not a `Functional` instance'
                             ''.format(func))
 
-        scalar = func.range.element(scalar)
+        scalar = func.domain.field.element(scalar)
```

Comment:
This should be the same, since `func` is checked to be a `functional` on the line above, and for `functional` we have `range = domain.field`.